src/buffer.rs

```diff
@@ -0,0 +1,294 @@
+use crate::ctx::TryIntoCtx;
```

Comment:
I think this entire module could actually be a third party crate; it is a self-defined type which implements a foreign trait, so ti won't violate orphan rules. (assuming we merge your Pwrite/Pread changes).